File: leetcode/BinarySearch/leetcode-240/240.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null) {
            return false;
        }
        if (matrix.length == 0) {
            return false;
        }
        int row = 0, colume = matrix[0].length - 1;
        while (row < matrix.length && colume >= 0) {
            if (matrix[row][colume] == target) {
                return true;
            } else if (matrix[row][colume] < target) {
                row++;
            } else {
                colume--;
            }
        }
        return false;
    }
}

File: leetcode/BinarySearch/leetcode-240/240.go
func searchMatrix(matrix [][]int, target int) bool {
    if len(matrix) == 0 || len(matrix[0]) == 0 {
		return false
	}
	row, colume := 0, len(matrix[0]) - 1
	for row < len(matrix) && colume >= 0{
		if target == matrix[row][colume] {
			return true
		} else if (target < matrix[row][colume]) {
			row++
		} else {
			colume--
		}
	}
	return false
}